src/assets/assetService.ts

```
@@ -94,13 +102,14 @@ export default class AssetService {
               sync.emit('error', e);
               if (e instanceof AbortError) return reject(e);
             }
+          await this.cleanUpOldBinaries();
```

Trap, log, and ignore any errors that occur here, so that the cleanup doesn't interfere with any other update functionality.

---

I was doing it inside the inner most function `AssetService.cleanUpOldBinariesFromDir`, which is called for `appmap`, `java` and `scanner` folders separately. In order to cover symlink resolutions as well, now I moved the try catch logic to `AssetService.cleanUpOldBinaries` instead of the call site above, because `AssetService.cleanUpOldBinaries` is called from two different places.